streaming_jupyter_integrations/schema_view.py

```diff
@@ -69,26 +72,29 @@ def get_schema(self) -> SchemaRoot:
     def _build_catalog(self, catalog: Row) -> SchemaCatalog:
         catalog_name = catalog[0]
         self.st_env.execute_sql(f"USE CATALOG `{catalog_name}`")
-        databases = self.st_env.execute_sql("SHOW DATABASES").collect()
-        tree_databases = [self._build_database(catalog_name, database) for database in databases]
+
+        with ThreadPoolExecutor(self.parallelism) as executor:
+            databases = self.st_env.execute_sql("SHOW DATABASES").collect()
+            tree_databases = [executor.submit(self._build_database, catalog_name, database) for database in databases]
         return SchemaCatalog(catalog_name, tree_databases)
 
     def _build_database(self, catalog_name: str, database: Row) -> SchemaDatabase:
         database_name = database[0]
         self.st_env.execute_sql(f"USE `{database_name}`")
 
-        tables = [table[0] for table in self.st_env.execute_sql("SHOW TABLES").collect()]
-        views = [view[0] for view in self.st_env.execute_sql("SHOW VIEWS").collect()]
-        tree_tables = [self._build_table(catalog_name, database_name, table, table in views) for table in tables]
-
-        functions = self.st_env.execute_sql("SHOW USER FUNCTIONS").collect()
-        tree_functions = [SchemaFunction(function[0]) for function in functions]
+        with ThreadPoolExecutor(self.parallelism) as executor:
+            tables = [table[0] for table in self.st_env.execute_sql("SHOW TABLES").collect()]
+            views = [view[0] for view in self.st_env.execute_sql("SHOW VIEWS").collect()]
+            tree_tables = [executor.submit(self._build_table, catalog_name, database_name, table, table in views) for table in tables]
+            functions = self.st_env.execute_sql("SHOW USER FUNCTIONS").collect()
+            tree_functions = [SchemaFunction(function[0]) for function in functions]
 
         return SchemaDatabase(database_name, tree_tables, tree_functions)
 
     def _build_table(self, catalog_name: str, database_name: str, table_name: str, is_view: bool) -> SchemaTable:
-        columns = self.st_env.execute_sql(f"DESCRIBE `{catalog_name}`.`{database_name}`.`{table_name}`").collect()
-        tree_columns = [self._build_column(column) for column in columns]
+        with ThreadPoolExecutor(self.parallelism) as executor:
+            columns = self.st_env.execute_sql(f"DESCRIBE `{catalog_name}`.`{database_name}`.`{table_name}`").collect()
+            tree_columns = [executor.submit(self._build_column, column) for column in columns]
```

Comment:
Do not parallelise build_column. Please note that execution of this method is almost instant. There is no point in parallelising it. I expect that making it parallel gives more overhead than benefits.